google_vision_python/main.py

```diff
@@ -0,0 +1,170 @@
+        if data.get('bucket').get('source'):
+            file_name = data.get('bucket').get('source').get('file')
+            source_path = data.get('bucket').get('source').get('path')
+        if data.get('bucket').get('destination'):
+            destination_path = data.get('bucket').get('destination').get('path')
+            bucket_access_token = data.get('bucket').get('destination').get('accessToken')
+            bucket_name = data.get('bucket').get('destination').get('bucketName')
+   
+    if data.get('response'):
+        topic_name = data.get('response').get('topic')
+
+    
+    bucket_credentials = google.oauth2.credentials.Credentials(bucket_access_token)
+
+    source_file_path = str(source_path) + '/' + str(file_name)
+
+    logging.error("Downloading file from bucket")
+    file_details = download_from_bucket(bucket_name, file_name, project_id, bucket_credentials, source_file_path)
+
+    logging.error("Checking if image is downloaded completely or not")
+    if file_details[1] == (os.stat(file_details[0]).st_size):
+        logging.info("File size match")
+
+        logging.error("Running vision API on image")
+        with io.open(file_details[0], 'rb') as image:
+            image_data = image.read()
+        vision_api_json_response = run_vision_api_on_image(image_data)
+
+        file_name_without_extension = os.path.splitext(file_name)[0]
+        file_name_json_format = str(file_name_without_extension) + '.json'
+
+        logging.error("Uploading vision api response to bucket")
+        upload_to_bucket(vision_api_json_response, file_name, source_path, bucket_name, project_id, bucket_credentials)
+                
+        logging.error("Creating custom pub-sub object")
+        pubsub_object = create_pubsub_object(uuid, variant_id, destination_path, file_name_json_format, constants.success_response)
+
+        logging.error(pubsub_object)
+        logging.error("Publishing custom pub-sub object to topic")
+        publish_to_pubsub_topic(constants.pubsub_project_id, topic_name, pubsub_object)
+
+        logging.error("Deleting files from temporary directory")
+        delete_files_from_directory(BASE_TEMP_DIR)
+
+        print('testing the response...')
+
+        return (constants.success_response, constants.success_status_code)
+    
+
+
+def create_pubsub_object(uuid, variant_id, file_path, file_name, status):
+    file_name_without_extension = os.path.splitext(file_name)[0]
+    file_name_json_format = str(file_name_without_extension) + '.json'
+    response_object = {
+        "uuid": uuid,
+        "variantId": variant_id,
+        "filePath" : file_path,
+        "fileName" : file_name_json_format,
+        "status": status
+    }
+    return response_object
+
+
+def publish_to_pubsub_topic(project_id, topic_name, data):
+    os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = constants.pubsub_key
+    publisher = pubsub_v1.PublisherClient()
+    logging.error(f'Publishing message to topic {topic_name}')
```

Comment:
It should be logging.info or use this

https://stackoverflow.com/questions/54147483/google-cloud-functions-python-logging-issue